packages/dds/tree/src/tableSchema.ts

```diff
@@ -902,7 +541,399 @@ export namespace TableSchema {
+		 * Gets the cell in the specified column, denoted by column ID.
+		 * @returns The cell if it exists, otherwise undefined.
+		 */
+		getCell(columnId: string): TreeNodeFromImplicitAllowedTypes<TCellSchema> | undefined;
+
+		/**
+		 * Sets the cell in the specified column.
+		 * @remarks To remove a cell, call {@link TableSchema.IRow.(removeCell:1)} instead.
+		 */
+		setCell(
+			column: IColumn,
+			value: InsertableTreeNodeFromImplicitAllowedTypes<TCellSchema>,
+		): void;
+		/**
+		 * Sets the cell in the specified column, denoted by column ID.
+		 * @remarks To remove a cell, call {@link TableSchema.IRow.(removeCell:2)} instead.
+		 */
+		setCell(
+			columnId: string,
+			value: InsertableTreeNodeFromImplicitAllowedTypes<TCellSchema>,
+		): void;
+
+		/**
+		 * Removes the cell in the specified column.
+		 * @privateRemarks TODO: return removed cell
+		 */
+		removeCell(column: IColumn): void;
+		/**
+		 * Removes the cell in the specified column, denoted by column ID.
+		 * @privateRemarks TODO: return removed cell
+		 */
+		removeCell(columnId: string): void;
+
+		/**
+		 * The row's properties.
+		 * @remarks This is a user-defined schema that can be used to store additional information about the row.
+		 * @privateRemarks
+		 * Note: these docs are duplicated on the inline type definitions in {@link createColumn}.
+		 * If you update the docs here, please also update the inline type definitions.
+		 */
+		get props(): TreeFieldFromImplicitField<TPropsSchema>;
+		set props(value: InsertableTreeFieldFromImplicitField<TPropsSchema>);
+	}
+
+	/**
+	 * Factory for creating new table column schema.
+	 * @internal
+	 */
+	export function createRow<
+		const TInputScope extends string | undefined,
+		const TCellSchema extends ImplicitAllowedTypes,
+	>(
+		inputSchemaFactory: SchemaFactoryAlpha<TInputScope>,
+		cellSchema: TCellSchema,
+	): ReturnType<
+		typeof System_TableSchema.createRowInternal<
+			TInputScope,
+			TCellSchema,
+			FieldSchema<FieldKind.Optional, typeof SchemaFactoryAlpha.null>
+		>
+	>;
+	/**
+	 * Factory for creating new table column schema.
+	 * @internal
+	 */
+	export function createRow<
+		const TInputScope extends string | undefined,
+		const TCellSchema extends ImplicitAllowedTypes,
+		const TPropsSchema extends ImplicitFieldSchema,
+	>(
+		inputSchemaFactory: SchemaFactoryAlpha<TInputScope>,
+		cellSchema: TCellSchema,
+		propsSchema: TPropsSchema,
+	): ReturnType<
+		typeof System_TableSchema.createRowInternal<TInputScope, TCellSchema, TPropsSchema>
+	>;
+	/** `createRow` implementation */
+	export function createRow(
+		inputSchemaFactory: SchemaFactoryAlpha,
+		cellSchema: ImplicitAllowedTypes,
+		propsSchema: ImplicitFieldSchema = SchemaFactory.optional(SchemaFactory.null),
+	): TreeNodeSchema {
+		return System_TableSchema.createRowInternal(inputSchemaFactory, cellSchema, propsSchema);
+	}
+
+	// #endregion
+
+	// #region Table
+
+	/**
+	 * A key to uniquely identify a cell in a table.
+	 * @sealed @internal
```

Comment:
not for this PR, but I think this needs to not be sealed to allow for its intended use. Sealed is for outputs, and this really should be marked input (which we don't have a way to do)

Reply:
Ah, yeah, good catch. Will update in a separate PR.